.github/workflows/build-test.yml

```diff
@@ -7,74 +7,114 @@
 #  Author: Michael E. Tryby
 #          US EPA - ORD/CESER
 #
+#          Caleb A. Buahin
+#          Xylem Inc.
+#
 
 name: Build and Test
 
 on:
   push:
-    branches: [ develop ]
+    branches: [develop]
   pull_request:
-    branches: [ develop ]
+    branches: [develop]
 
 jobs:
-  win_build:
-    runs-on: windows-2016
+  build:
+    runs-on: ${{ matrix.os }}
     continue-on-error: ${{ matrix.experimental }}
-
-    defaults:
-      run:
-        shell: cmd
-
+    env:
+      OMP_NUM_THREADS: 1
+      PROJECT: swmm
+      BUILD_HOME: build
+      TEST_HOME: nrtests 
+      NRTESTS_URL: https://github.com/OpenWaterAnalytics/swmm-nrtestsuite
     strategy:
       fail-fast: false
       matrix:
-        generator: [Visual Studio 15 2017]
-        experimental: [false]
+        os: [windows-2016, ubuntu-latest, macos-latest]
         include:
-          - generator: Visual Studio 15 2017 Win64
+          - os: windows-2016
+            generator: Visual Studio 15 2017 Win64
```

Comment:
We can set preferred toolset in CMakeLists.txt and use default generator here as well. 

https://cmake.org/cmake/help/v3.8/generator/Visual%20Studio%2015%202017.html

Reply:
Not sure I understand the first part of what you are saying here. I have used the default generator though.